### DISCUSSION TITLE
Allow linking to identifiers in code

### DISCUSSION THREAD

**Discussion by CAFxX** [community/community#11467] [Other Feature Feedback, Questions, & Ideas]
Currently Github allows to [link to specific lines in a file](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet). This is normally useful only in the context of a permalink to a specific commit: without the permalink as code is added/removed the linked-to lines move around and the link loses its usefulness.

It would be ideal to allow users to also link to identifiers in a file, instead of specific lines. E.g. in this Go snippet (main.go):

```go
package main

func main() {
}
```

a user could create a link of the form `github.com/user/repo/blob/master/main.go#main` that would resolve to lines 3-4 of the snippet above (i.e. to the whole body of the `main` function). Later, if main.go is modified as in the snippet below, the same link would instead resolve to lines 5-8:

```go
package main

import "fmt"

func main() {
  // Greet the user
  fmt.Println("Hello world!")
}
```

If the identifier in the URL is not found in the file, a warning could be shown at the top of the page, potentially instructing to follow a link to perform a code search for the identifier across the whole repo.

Identifiers should not be limited to just functions, but also to global variables, types/classes/structs, etc.

**Comment by Sysix** [comment]
I had multiple times the same problem too. Would love to get a user friendly fix for it.

My Idea: When Copy Pasting a link from a active / main branch, auto replace the url part with the current commit hash.
